openfe/utils/visualization.py

```diff
@@ -140,22 +157,49 @@ def draw_mapping(mol1_to_mol2: Dict[int, int],
     if not d2d:
         d2d = Chem.Draw.rdMolDraw2D.MolDraw2DCairo(600, 300, 300, 300)
 
-    # Need to squash to 2D
-    mol1_copy = copy.deepcopy(mol1)
-    mol2_copy = copy.deepcopy(mol2)
-    AllChem.Compute2DCoords(mol1_copy)
-    AllChem.Compute2DCoords(mol2_copy)
+    return _draw_molecules(
```

Comment:
So thought here - since we'll possibly use py3dmol in the future to get 3D mapping visualisations (could be fun), do we need to consider specifying the backend here (or at least opening up the idea that we could switch between backends?)

Probably should be a one molecule mapping thing though (not sure you can do side-by-side comparison in py3dmol)

Reply:
I was thinking something like that might be a change to the `_draw_molecules`; maybe rename `d2d` as `renderer` and then `_draw_molecules` can select between `_render_drawmol2d` and `_render_py3dmol` depending on what it receives. My main goal was to move most of the drawing stuff into a separate method so we separate the stuff that is fundamentally our responsibility (decided which atoms/bonds to highlight/color) from the stuff that is just using rdkit's drawing tools to render.

I think this argues for moving the default d2d creation into the `_draw_molecules`. I've done that in [ed3fb45](https://github.com/OpenFreeEnergy/openfe/pull/100/commits/ed3fb45fb112bb4ad29d134f482f311b283f1db8)